### PR TITLE
add datetime, date, and decimal encoders

### PR DESCRIPTION
**Pull request by daveauerbach**

### CODE REVIEW

**Comment by daveauerbach** [comment]
@wonga00 @cvanderschuere

File: tests/test_nonblocking_stream.py [hunk 1]
```diff
@@ -70,9 +84,21 @@ def test_send_event(self):
...
 
+    def test_send_hard_to_encode_data(self):
+        s = nonblocking_stream.NonblockingStream('test_stream', 'pkey')
+        test_data = generate_messy_test_data()
+        s.put(**test_data)
+        meta_data, message_data = generate_transmitted_record(test_data)
+        mock_sent_meta_data, mock_sent_message_data = (
+            nonblocking_stream.threadLocal
+            .zmq_socket.send_multipart.calls[0][0][0])
+        assert_equal(mock_sent_meta_data, meta_data)
```
Comment (wonga00) [inline tests/test_nonblocking_stream.py]: shouldn't we check that we are sending the encoded types?

Comment (daveauerbach) [inline tests/test_nonblocking_stream.py]: Not sure what you mean. This test will fail if the types weren't being encoded.

Comment (wonga00) [inline tests/test_nonblocking_stream.py]: yeah, we should check though they are encoded to the right value. like seconds vs isostring right?

**Comment by daveauerbach** [comment]
@wonga00 r?

**Comment by wonga00** [comment]
r+

**Pull request closed without merging**